### PR TITLE
Scrape

### PR DESCRIPTION
**Pull request by benigeri**

### CODE REVIEW

**Comment by antoniocarlosortiz** [comment]
@benigeri 
I got it working!

**Comment by benigeri** [comment]
@antoniocarlosortiz wooo!!!! That's some pretty next level coding skills huh?

**Comment by antoniocarlosortiz** [comment]
@benigeri For your review :D

**Comment by benigeri** [comment]
Lots of commits! @antoniocarlosortiz can you please rebase and clean up your commits?  1 commit is probably fine for this. Please ping me once done and I can take a look at it first thing tomorrow am.

**Comment by antoniocarlosortiz** [comment]
@benigeri hi paul. Rebase already done. I run celerybeat inside the scrape folder with:
```celery --workdir=../ -A tasks worker --loglevel=info --beat```
I use workdir so the scraper and celery can read dependencies on other folders that is located on root.

**Comment by benigeri** [comment]
@antoniocarlosortiz you need to fix the tests

File: scrape/nodes_and_edges/pipelines.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# -*- coding: utf-8 -*-
+
+# Define your item pipelines here
+#
+# Don't forget to add your pipeline to the ITEM_PIPELINES setting
+# See: http://doc.scrapy.org/en/latest/topics/item-pipeline.html
+<<<<<<< HEAD
+<<<<<<< HEAD
+<<<<<<< HEAD
```
Comment (benigeri) [inline scrape/nodes_and_edges/pipelines.py]: there are a lot of issues with your rebase... what happened?

File: .gitignore [hunk 2]
```diff
@@ -48,3 +48,9 @@ tests/test_employee_counts.py
 *.swp
 *.swo
 *.swc
+
+# Credentials
+*credentials.py
```
Comment (benigeri) [inline .gitignore]: make sure to remove this line

File: default_config.ini [hunk 3]
```diff
@@ -24,3 +24,7 @@ engine_args_literal = {
     'echo': False,
     'client_encoding': 'utf8'
   }
+
+[scrape]
+session_key =
```
Comment (benigeri) [inline default_config.ini]: rename to linkedin username/password

File: requirements.txt [hunk 4]
```diff
@@ -11,3 +11,8 @@ psycopg2==2.6
 SQLAlchemy==1.0.4
 Werkzeug==0.10.4
 WTForms==2.0.2
+Scrapy==0.24.6
```
Comment (benigeri) [inline requirements.txt]: remove scrapy and remember to update requirements.in

File: scrape/tasks.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+celery.config_from_object('celeryconfig')
+
+
+@celery.task
+def run_spider():
+    company_list = companies_controller.get_companies()
+    item_list = []
+    for company in company_list:
+        url = company.url
+        pk = company.company_id
```
Comment (benigeri) [inline scrape/tasks.py]: what is pk? call this company_id

File: scrape/tasks.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+        item_tuple = ScraperScript(url)
+        item = item_tuple
```
Comment (benigeri) [inline scrape/tasks.py]: why don't you just do 

```
item = ScraperScript(url)
```

also item isn't the best name, maybe `result`

File: scrape/tasks.py (hunk 5 above)
Comment (antoniocarlosortiz) [inline scrape/tasks.py]: its supposed to be primary key but yeah. People always ask what that is so i better change it.

**Comment by benigeri** [comment]
@antoniocarlosortiz good to merge

**Comment by benigeri** [comment]
:)

**Comment by antoniocarlosortiz** [comment]
its so fulfilling!!

**Pull request merged**